runtime/src/main/java/org/corfudb/runtime/collections/ProtobufIndexer.java

```diff
@@ -0,0 +1,74 @@
+package org.corfudb.runtime.collections;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import com.google.protobuf.Message;
+
+import org.corfudb.common.util.ClassUtils;
+import org.corfudb.runtime.CorfuOptions;
+import org.corfudb.runtime.collections.CorfuTable.Index;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+/**
+ * This layer implements the
+ * ProtobufIndexer uses the special FieldOptions that the application can place on its
+ * protobuf definitions (like secondary_key) and create secondary indexes callbacks over CorfuTable
+ * based on that.
+ *
+ * Created by hisundar on 2019-08-12.
+ */
+public class ProtobufIndexer implements CorfuTable.IndexRegistry<Message, CorfuRecord<Message, Message>> {
+
+    private HashMap<String,
```

Comment:
I'm worried about race condition here

Reply:
indexes are created only at the time of table creation and only read afterwards. so there is no concurrent mutations that can result in a race condition here right?

Reply:
yep